### PR TITLE
Implement raw version of subscription history

### PR DESCRIPTION
**Pull request by sanjay900**
While thinking about #11, i realised that we would want a raw version of the subscription history events as well, so I have split the history parser up a little to allow reusing that logic for subscriptions as well.
This also means things won't crash if something goes wrong while parsing subscription history events.

### CODE REVIEW

**Comment by sanjay900** [comment]
I was able to validate this on my b series panel by initiating a panel sensor reset, and seeing that event come through

File: bosch_alarm_mode2/history.py [hunk 1]
```diff
@@ -19,6 +19,14 @@ class HistoryEvent(NamedTuple):
     def __repr__(self):
         return f"[{self.id}] {self.date} | {self.message}"
 
+class HistoryEventParams(NamedTuple):
+    date: datetime
+    event_code: int
```
Comment (mag1024) [inline bosch_alarm_mode2/history.py:24]: We actually seem to treat this as str in most places

File: bosch_alarm_mode2/history.py [hunk 2]
```diff
@@ -66,12 +74,16 @@ def parse_polled_events(self, event_data):
 
     def parse_subscription_event(self, raw_event):
         text_len = BE_INT.int16(raw_event, 23)
+        event_id = BE_INT.int32(raw_event)
```
Comment (mag1024) [inline bosch_alarm_mode2/history.py:77]: should probably length check raw_event -- or put the try around the whole thing

File: bosch_alarm_mode2/history.py [hunk 3]
```diff
@@ -66,12 +74,16 @@ def parse_polled_events(self, event_data):
...
-        date = datetime.strptime(f"{date} {time}", "%m/%d/%Y %I:%M%p")
-        self._events.append((BE_INT.int16(raw_event, 4), date, message))
+        try:
+            e = self._parser.parse_subscription_event(raw_event)
+            LOG.debug(e)
+            self._events.append(e)
+        except Exception as excp:
+            error_str = f"parse error: {repr(excp)}"
+            LOG.error("History event " + error_str)
+            self._events.append(HistoryEvent(event_id + 1, datetime.now(), error_str))
```
Comment (mag1024) [inline bosch_alarm_mode2/history.py:86]: Share this with parse_polled_events?

**Review by mag1024 — APPROVED** [review]
Yeah, good idea!
Let's get drop the TextHistory next, and we can probably remove the abstractmethods as well.

**Comment by sanjay900** [comment]
Yeah, i like the idea of having a commit around that has all the text stuff implemented, before we drop it just in case there is ever a point we need it all working again for some reason

**Comment by mag1024** [comment]
Thanks!
Next time we touch this file, we should wrap some of the extra long lines -- pylint should point them out.

**Pull request merged**